cpp/src/parquet/platform.h

```diff
@@ -24,7 +24,8 @@
 #include "arrow/io/interfaces.h"  // IWYU pragma: export
 #include "arrow/status.h"         // IWYU pragma: export
 #include "arrow/type_fwd.h"       // IWYU pragma: export
-#include "arrow/util/macros.h"    // IWYU pragma: export
+#include "arrow/util/compression.h"
```

Comment:
I don't think adding this inclusion is necessary, right?

Reply:
If not, then it will prevent us to initialize as `CodecOptions{}`? Am I doing this right?
```
column_writer.h:101:43: error: invalid use of incomplete type 'parquet::CodecOptions' (aka 'arrow::util::CodecOptions')
      const CodecOptions& codec_options = CodecOptions{});
                                          ^~~~~~~~~~~~~~
```

Reply:
Maybe forward declr like `default_memory_pool()` can be used?

Reply:
@yyang52 You don't need it in `platform.h`.

Reply:
yes, could I include that directly in `column_writer.h`, as the forward declaration is not sufficient to initialize an object as far as I learned?